### DISCUSSION TITLE
Lessson 7: Arguments in getConversionRate() function

### DISCUSSION THREAD

**Discussion by cell0901** [Cyfrin/foundry-full-course-cu#1456] [General]
In Lesson 7 around 37:10

here the function is taking two argument the ``ethAmount`` and  ``priceFeed`` and its working fine 
```
 function getConversionRate(
        uint256 ethAmount,
        AggregatorV3Interface priceFeed
    ) internal view returns (uint256) {
        uint256 ethPrice = getPrice(priceFeed);
        uint256 ethAmountInUsd = (ethPrice * ethAmount) / 1000000000000000000;
        // the actual ETH/USD conversion rate, after adjusting the extra 0s.
        return ethAmountInUsd;
    }
}
```

But as soon i replace the ``ethAmount`` to second place as an argument and ``priceFeed`` to first place the code gives me this error

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/142923689/76c9e610-a461-4de5-989a-ddb36c50f3c7)
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/142923689/eae3703c-d66c-4171-86d7-930d66de36e0)

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/142923689/d5eaa9da-b411-490b-8ed4-03742ab2e23d)

**Comment by EngrPips** [comment]
Hello @cell0901 , It makes total sense that you are getting that error and i would explain why it makes to you. 

First of all lets remind ourselves the fact that  the function is inside of a library and a library in solidity would always take in the type that it is meant to be used on as the first argument. If you recall in the FundMe.sol contract you declared something like `using priceConverter for uint256`, declaring such means you would be calling any function in that library with a `uint256` as the first argument. But as soon you replace the `ethAmount` to second place as an argument and priceFeed to first place as an argument, that means you are intending to use the library on the type of `AggregatorV3Interface` and not type of `uint256` but then you tried to call the library on type of uint256 again `require(msg.value.getConversionRate(S_priceFeed) >= MINIMUM_USD, "You need to spend more ETH");`, In the library uint256 is not more the first argument so it would return you that error as shown above 
`Member "getConversionRate" not found or not visible after argument-dependent lookup in uint256`.

I hope ecplained well enough for you to understand

**Reply by cell0901** [reply]
Oh now i understand, thanks for helping !

**Reply by cell0901** [reply]
how can i mark it as answer? there is no option i see

**Reply by EngrPips** [reply]
Hello @cell0901  i assume if you click on that 3 dots `...` on my comment that you want to mark as an answer, it should pop a menu and in it there should an option to mark the comment as answer
